src/azul/__init__.py

```diff
@@ -518,27 +532,29 @@ def is_internal(self):
             #        https://github.com/DataBiosphere/azul/issues/2865
             return self.is_integration_test_catalog or self.name == 'dcp3'
 
+        @classmethod
+        def from_json(cls, name: str, spec: JSON) -> 'Config.Catalog':
+            return cls(name=name, **spec)
+
+        def to_dict(self):
+            return {'name': self.name, 'atlas': self.atlas, 'plugins': self.plugins}
+
     @cached_property
     def catalogs(self) -> Mapping[CatalogName, Catalog]:
         """
         A mapping from catalog name to a mapping from plugin type to plugin
         package name.
         """
-        catalogs = os.environ['AZUL_CATALOGS'].split(',')
-        catalogs = {
-            catalog: self.Catalog(
-                name=catalog,
-                atlas=atlas,
-                plugins={
-                    plugin_type: plugin
-                    for plugin_type, plugin in map(splitter('/'), plugins)
-                }
-            )
-            for atlas, catalog, *plugins in map(splitter(':'), catalogs)
-        }
+        catalogs = {}
+        # FIXME: Import locally to avoid conflicts with `azul.json` module
+        #        during unit tests.
+        #        https://github.com/DataBiosphere/azul/issues/3133
+        import json
+        for name, catalog_spec in json.loads(os.environ['AZUL_CATALOGS']).items():
+            c = self.Catalog.from_json(name, catalog_spec)
+            assert name not in catalogs
```

Comment:
That's impossible, assuming `dict` works correctly. This assert is preventing you from using a much simpler dictionary comprehension AND it doesn't add clarity or express and otherwise hidden invariant.

Reply:
```diff
Index: src/azul/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/__init__.py b/src/azul/__init__.py
--- a/src/azul/__init__.py	(revision 1de54f360d1db0c3b1173bd491ef4b559d08b741)
+++ b/src/azul/__init__.py	(date 1624324345853)
@@ -545,17 +545,15 @@
         A mapping from catalog name to a mapping from plugin type to plugin
         package name.
         """
-        catalogs = {}
-        # FIXME: Import locally to avoid conflicts with `azul.json` module
-        #        during unit tests.
+        # FIXME: Eliminate local import
         #        https://github.com/DataBiosphere/azul/issues/3133
         import json
-        for name, catalog_spec in json.loads(os.environ['AZUL_CATALOGS']).items():
-            c = self.Catalog.from_json(name, catalog_spec)
-            assert name not in catalogs
-            catalogs[name] = c
+        catalogs = json.loads(os.environ['AZUL_CATALOGS'])
         require(bool(catalogs), 'No catalogs configured')
-        return catalogs
+        return {
+            name: self.Catalog.from_json(name, catalog)
+            for name, catalog in catalogs.items()
+        }
 
     @cached_property
     def default_catalog(self) -> CatalogName:
```